### ISSUE TITLE
idea: flux-archive: support archive of an executable and its libraries

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#5753]
The Slurm sbcast(1) tool supports a `--send-libs` option which broadcasts an executable and a possible subset of its libraries to the nodes of a job.

It would probably be trivial to extend `flux-archive(1)` to support archival of an executable and libs in the kvs. The shell stage-in plugin could somehow be made cognizant that dependent libraries are archived along with an executable and could automatically prepend to `LD_LIBRARY_PATH` so the local copy of libs is used.

It may be useful to support exclusion of some directories for automated library archiving, i.e. if they are already node local, so some kind of config might be useful.

**Comment by garlick** [comment]
I wonder if somehow libtool could be employed here so the archive would include a libtool wrapper for the executable that sets up paths properly to find local copies of libraries (e.g. from the archive?)

**Comment by garlick** [comment]
While briefly considering this and looking for tools I of course got into container land fairly quickly.   Since I'm not very familiar with the tooling in this space, I'm not sure if the answer is "just use a container" here or if this simple extension to `flux archive` would indeed be useful.

The first thing I looked at was https://apptainer.org/ and quickly discovered
- It the new fork of singularity, and got in the weeds reading about the kerfuffle that @vosch had mentioned earliler.
- [LC supports singularity](https://hpc.llnl.gov/services/cloud/containers).
- I spent 15 minutes bumbling around trying to build it on my test cluster, and then another squirrel came along.

Anyway.  I think I lack perspective to decide if it's useful to go down this path so others should weigh in.

Also we should look at the LC container guide and see if there are any TODOs evident there for flux.
